openpype/hosts/maya/plugins/publish/validate_gltf_textures_names.py

```diff
@@ -0,0 +1,130 @@
+        if invalid:
+            raise RuntimeError("Non PBS material found in "
+                               "{0}".format(invalid))
+        invalid = self.get_texture_node_invalid(instance)
+        if invalid:
+            raise RuntimeError("Related texture file "
+                               "nodes not connected")
+        invalid = self.get_texture_name_invalid(instance)
+        if invalid:
+            raise RuntimeError("Invalid texture name(s) found: "
+                               "{0}".format(invalid))
+
+    def get_texture_name_invalid(self, instance):
+
+        invalid = set()
+        shading_grp = self.shader_selection(instance)
+
+        # get the materials related to the selected assets
+        # get the file textures related to the PBS Shader
+        # validate the names of the textures
+        for material in shading_grp:
+            main_shader = cmds.listConnections(material,
+                                               destination=True,
+                                               type="StingrayPBS")
+            for shader in main_shader:
+                albedo = cmds.listConnections(shader + ".TEX_color_map")[0]
+                dif_path = cmds.getAttr(albedo + ".fileTextureName")
+                dif = dif_path.split(".")[0]
+                # "_D"
+                if not dif.endswith("_D"):
+                    invalid.add(dif_path)
+                orm_packed = cmds.listConnections(shader + ".TEX_ao_mapX")[0]
+                # "_ORM"
+                orm_path = cmds.getAttr(orm_packed + ".fileTextureName")
+                orm = orm_path.split(".")[0]
+                if not orm.endswith("_ORM"):
+                    invalid.add(orm_path)
+                nrm = cmds.listConnections(shader + ".TEX_normal_map")[0]
+                nrm_path = cmds.getAttr(nrm + ".fileTextureName")
+                nrm_map = nrm_path.split(".")[0]
+                # "_N"
+                if not nrm_map.endswith("_N"):
+                    invalid.add(nrm_path)
+
+        return list(invalid)
+
+    def get_texture_node_invalid(self, instance):
+        invalid = set()
+        shading_grp = self.shader_selection(instance)
+        for material in shading_grp:
+            main_shader = cmds.listConnections(material,
+                                               destination=True,
+                                               type="StingrayPBS")
+            for shader in main_shader:
+                # diffuse texture file node
+                albedo = cmds.listConnections(shader + ".TEX_color_map")
+                if not albedo:
+                    invalid.add(albedo)
+                orm_packed = cmds.listConnections(shader + ".TEX_ao_mapX")
+                if not orm_packed:
+                    invalid.add(orm_packed)
+                nrm = cmds.listConnections(shader + ".TEX_normal_map")
+                if not nrm:
+                    invalid.add(nrm)
```

Comment:
I think this is causing that if you don't have texture pack connected - ie. all of these file nodes to `TEX_color_map`, `TEX_ao_mapX`, `TEX_normal_map` then it will fail. My testing case is simple shader with only one diffuse texture `foo_D.png` connected to `TEX_color_map` and it is triggering this validator with `Related texture file nodes not connected`.

Reply:
Also, it's adding `None` into `invalid`

I'd also, for these cases explicitly check for input connections (source=True) and disregard outgoing connections (destination=False). Rather be exact and explicit about what we're checking :)